src/components/ColorPicker/index.tsx

```diff
@@ -0,0 +1,157 @@
+                        onClick={() => {
+                            setColor(initialColor);
+                        }}
+                    >
+                        {" "}
+                    </div>
+                    <p> CURRENT </p>
+                </div>
+                <div className={styles.oldColorContainer}>
+                    <div
+                        className={styles.newColor}
+                        style={{ backgroundColor: color }}
+                    >
+                        {" "}
+                    </div>
+                    <p> NEW </p>
+                </div>
+                <div className={styles.oldColorContainer}>
+                    <HexColorInput
+                        className={styles.hexInput}
+                        color={color}
+                        onChange={setColor}
+                    />
+                    <p> HEX </p>
+                </div>
+            </div>
+            <div
+                className={classNames([
+                    styles.colors,
+                    styles.agentColorsSwatches,
+                ])}
+            >
+                {AGENT_COLORS.map((color) => (
+                    <Tooltip
+                        key={color}
+                        align={{ offset: [2, 5] }}
+                        color="#69738A"
+                        overlayClassName={styles.tooltip}
+                        title={color.slice(1)}
+                    >
+                        <button
+                            key={color}
+                            className={styles.swatch}
+                            style={{ background: color }}
+                            onClick={() => setColor(color)}
+                        />
+                    </Tooltip>
+                ))}
+            </div>
+            <p className={styles.recentColorText}> Recent </p>
+            <div className={styles.colors}>
+                {recentColors.map((color) => (
+                    <button
+                        key={color}
+                        className={styles.swatch}
+                        style={{ background: color }}
+                        onClick={() => setColor(color)}
+                    />
+                ))}
+            </div>
+        </div>
+    );
+
+    return (
+        <Popover
+            overlayClassName={styles.popover}
+            open={isOpen}
+            content={renderColorPickerComponent()}
+            placement="right"
+            onOpenChange={closePopover}
+            trigger="click"
+        />
+    );
+};
+
+const mapStateToProps = (state: any) => ({
+    recentColors: selectionStateBranch.selectors.getRecentColors(state),
+});
+
+const dispatchToPropsMap = {
+    setColorChanges: selectionStateBranch.actions.setColorChanges,
+    setRecentColors: selectionStateBranch.actions.setRecentColors,
+};
+
+export default connect(mapStateToProps, dispatchToPropsMap)(ColorPicker);
```

Comment:
this shouldn't be connected to state if it's a component (ie in the component folder). I thought we had discussed a way to send down this information without needing this connection